documentation/binding-data/tutorial-flow-components-binder-beans.asciidoc

```diff
@@ -86,37 +98,59 @@ binder.forField(yearOfBirthField)
   .bind(Person::getYearOfBirth, Person::setYearOfBirth);
 ----
 
-Note that to use Bean Validation annotations you should have a JSR 303 implementation like Hibernate Validator available in your classpath. E.g. if you use Maven, the following dependency can be used.
+==== Defining Constraints on the Bean Level
 
-[source, xml]
+// The original text was difficult to understand. I copied this code example from the Hibernate docs and referenced them as the source. The original text just linked to the example. We could leave it like this or amend the code. I'm very unsure about the explanation here.   
+
+As an alternative to defining constraint annotations for specific properties, you can define constraints on the bean level.
+
+*Example*: Defining bean-level constraint annotations. _Source: https://docs.jboss.org/hibernate/validator/5.4/reference/en-US/html_single/#validator-gettingstarted-createmodel[Hibernate Validator 5.4.2 - Applying constraints]_
+
+[source, java]
 ----
-<dependency>
-    <groupId>org.hibernate</groupId>
-    <artifactId>hibernate-validator</artifactId>
-    <version>5.4.1.Final</version>
-</dependency>
+package org.hibernate.validator.referenceguide.chapter01;
+
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+public class Car {
+
+    @NotNull
+    private String manufacturer;
+
+    @NotNull
+    @Size(min = 2, max = 14)
+    private String licensePlate;
+
+    @Min(2)
+    private int seatCount;
+
+    public Car(String manufacturer, String licencePlate, int seatCount) {
+        this.manufacturer = manufacturer;
+        this.licensePlate = licencePlate;
+        this.seatCount = seatCount;
+    }
+
+    //getters and setters ...
+}
 ----
 
-Constraint annotations can also be defined on the bean level instead of being defined for any specific property.
```

Comment:
@pleku Do we nowadays support bean level JSR 303 validators ? If not (as I have thought), this comment is really confusing here. I guess that is why @ruthstocks re-wrote this part into something that it is today.